src/bin/main.rs

```diff
@@ -464,6 +472,14 @@ fn determine_operation(matches: &Matches) -> Result<Operation, OperationError> {
         if minimal_config_path.is_some() {
             return Err(OperationError::MinimalPathWithStdin);
         }
+        if matches.opt_present("check") {
```

Comment:
Would it be better to move these checks into `format_string` (https://github.com/rust-lang/rustfmt/blob/master/src/bin/main.rs#L241) so that we can use the already parsed values? (`options.check` and `emit_mode`)

Reply:
That sounds reasonable, yes - it didn't feel quite right comparing with a literal there! I'll take a look there, thanks.

Reply:
I've moved the check into `format_string`.  I'm still returning `Err(OperationError)` but with added `.into()` to convert to `FailureError` - maybe `?` would be better?